src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -419,32 +419,26 @@ private static boolean isInSwitchRule(DetailAST ast) {
     }
 
     /**
-     * Checks if current expression is a switch labeled expression. If so,
-     * braces are not allowed e.g.:
-     * <p>
-     * {@code
-     * case 1 -> 4;
+     * Checks if the current case or default label is a switch labeled expression.
+     * If so, braces are not allowed e.g.:
+     * <pre>
+     * return switch(x) {
+     *    case Integer i -&gt; 3; // no braces allowed
```

Comment:
Braces are allowed if you add a yield:
````
		return switch (x) {
		case Integer i -> {
			yield 3;
		}
````

Isn't this the same with lambdas were you have to add a return if you specifically add the braces since it was implied before?

Reply:
> Braces are allowed if you add a yield:

Yes, but we should not enforce this code to have braces for a single return value like this. because if we want to fix this violation we will have to add **unnecessary** `yield`.

Reply:
> Isn't this the same with lambdas were you have to add a return if you specifically add the braces since it was implied before?

Yes actually
[
![2](https://github.com/user-attachments/assets/4200e6f0-1b4c-4ac8-8597-916f55aa7b80)
](url)

we can remove braces and return from lambda but we will have a check violation
The check doesn't allow us to use the expression lambda. I don't think it is a good idea to violate this kind of code (code where we need to add unnecessary keywords such as return and yield to satisfy the check)

Reply:
If our check supports lambdas and does the same enforcement with lambdas and needing to write `return`s, then I question why this issue is valid for switches but not for lambdas.
https://checkstyle.org/checks/blocks/needbraces.html#Example6-config

I didn't provide comments on approval of the issue, so I am now asking to reconsider why this issue is valid.
@nrmancuso

Reply:
The reason we added lambda support was to make it clear where the lambda started and ended if it spanned more than a single line.

Reply:
Check behavior should be consistent between lambdas/blocks/returns and switch rule/blocks/yields IMO

Reply:
```
D:\CS\test
javac src/Test.java
D:\CS\test
cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NeedBraces">
      <property name="tokens" value="LAMBDA, LITERAL_CASE"/>
    </module>
  </module>
</module>
D:\CS\test
cat src/Test.java
import java.util.Arrays;
import java.util.List;

public class Test {
    int test(Object o, String z) {
        List<Integer> numbers = Arrays.asList(1, 2, 3);
        numbers = numbers.stream()
                .map(x -> 0) // violation
                .toList();

        numbers = numbers.stream()
                .map(x -> z.length()) // violation
                .toList();

        return switch (o) {
            case Integer _ -> 5; // should be violation same as line below and lambda examples above
            case String s -> s.length(); // violation
            default -> 0;
        };
    }
}
D:\CS\test
java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:8:24: '->' construct must use '{}'s. [NeedBraces]
[ERROR] D:\CS\test\src\Test.java:12:24: '->' construct must use '{}'s. [NeedBraces]
[ERROR] D:\CS\test\src\Test.java:17:13: 'case' construct must use '{}'s. [NeedBraces]
Audit done.
Checkstyle ends with 3 errors.
```

---

So, we are saying that line 16 (` case Integer _ -> 5;` ) is a false negative? note that in all cases we have to introduce unnecessary `return` or yield

Reply:
To me, that is what it sounds like. If someone doesn't want the violation they could turn on `allowSingleLineStatement`, create a suppression, or remove the token.

I think we should redo the issue first and confirm before we redo the PR.

> value="LAMBDA, LITERAL_CASE"

I believe I would expect a violation on the default tokens if that token was added too.

Reply:
I have reviewed and reapproved the issue, thanks a lot for catching this @rnveach